File: helfomat-web/src/test/java/de/helfenkannjeder/helfomat/web/controller/SearchControllerTest.java
package de.helfenkannjeder.helfomat.web.controller;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.MediaType;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;
import org.springframework.web.context.WebApplicationContext;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

/**
 * @author Valentin Zickner
 */
@ExtendWith(SpringExtension.class)
@SpringBootTest
public class SearchControllerTest {

    private MockMvc mockMvc;

    @Autowired
    private WebApplicationContext webApplicationContext;

    @BeforeEach
    public void setUp() {
        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
    }


    @Test
    void searchQuestions_withNoMoreInforation_returnsListOfQuestions() throws Exception {
        // Arrange


        // Act
        mockMvc.perform(get("/api/questions"))
            .andExpect(status().isOk())
            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
            .andExpect(jsonPath("$").isArray())
            .andExpect(jsonPath("$.length()").value(5))
            .andExpect(jsonPath("$[0]").isMap())
            // @formatter:off
            .andExpect(jsonPath("$[0].question").value("Möchtest Du Dich in medizinischer Hilfe ausbilden lassen und diese leisten?"))
            .andExpect(jsonPath("$[1].question").value("Du hast keine Angst vor Feuer, nur Respekt. Möchtest Du lernen, wie man es löscht?"))
            .andExpect(jsonPath("$[2].question").value("Hast Du Interesse daran, Menschen bei Verkehrsunfällen aus Fahrzeugen zu retten?"))
            .andExpect(jsonPath("$[3].question").value("Kannst Du sehr kurzfristig, ggf. auch innerhalb Deiner Arbeitszeiten, für Einsätze zur Verfügung stehen?"))
            .andExpect(jsonPath("$[4].question").value("Ist Kochen eine Deiner Leidenschaften?"));
            // @formatter:on
    }
}

File: helfomat-web/src/test/kotlin/de/helfenkannjeder/helfomat/web/controller/SearchControllerTest.kt
package de.helfenkannjeder.helfomat.web.controller

import de.helfenkannjeder.helfomat.core.ProfileRegistry
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.http.MediaType
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers

/**
 * @author Valentin Zickner
 */
@ExtendWith(SpringExtension::class)
@SpringBootTest
@AutoConfigureMockMvc
@ActiveProfiles(ProfileRegistry.TEST)
class SearchControllerTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    @Throws(Exception::class)
    fun searchQuestions_withNoMoreInformation_returnsListOfQuestions() {
        mockMvc.perform(MockMvcRequestBuilders.get("/api/questions"))
            .andExpect(MockMvcResultMatchers.status().isOk)
            .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
            .andExpect(MockMvcResultMatchers.jsonPath("$").isArray)
            .andExpect(MockMvcResultMatchers.jsonPath("$.length()").value(5))
            .andExpect(MockMvcResultMatchers.jsonPath("$[0]").isMap)
            .andExpect(MockMvcResultMatchers.jsonPath("$[0].question").value("Möchtest Du Dich in medizinischer Hilfe ausbilden lassen und diese leisten?"))
            .andExpect(MockMvcResultMatchers.jsonPath("$[1].question").value("Du hast keine Angst vor Feuer, nur Respekt. Möchtest Du lernen, wie man es löscht?"))
            .andExpect(MockMvcResultMatchers.jsonPath("$[2].question").value("Hast Du Interesse daran, Menschen bei Verkehrsunfällen aus Fahrzeugen zu retten?"))
            .andExpect(MockMvcResultMatchers.jsonPath("$[3].question").value("Kannst Du sehr kurzfristig, ggf. auch innerhalb Deiner Arbeitszeiten, für Einsätze zur Verfügung stehen?"))
            .andExpect(MockMvcResultMatchers.jsonPath("$[4].question").value("Ist Kochen eine Deiner Leidenschaften?"))
    }
}